pc/segmentation/regiongrowing/regiongrowing.go

```diff
@@ -0,0 +1,56 @@
+package regiongrowing
+
+import (
+	"github.com/seqsense/pcgol/mat"
+	"github.com/seqsense/pcgol/pc"
+	storage "github.com/seqsense/pcgol/pc/storage/kdtree"
+)
+
+const initialSliceCap = 8192
+
+type RegionGrowing struct {
+	*storage.KDTree
+	propertyIter pc.Uint32Iterator
+}
+
+func New(ra pc.Vec3RandomAccessor, propertyIter pc.Uint32Iterator) *RegionGrowing {
+	return &RegionGrowing{
+		KDTree:       storage.New(ra),
```

Comment:
It seems not KDTree specific.
Receiving `storage.Search` instead of `pc.Vec3RandomAccessor` would be better.